proposals/lip-0045.md

```diff
@@ -984,166 +942,357 @@ The Interoperability module contains the following commands:
+            "length": HASH_LENGTH, 
+            "fieldNumber": 1
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
 ```
 
-#### getChainAccount
+#### TerminatedStateAccountCreated
 
-The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_STATE_CREATED`. This event is emitted when a new entry is created in the terminated state substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated chain account.
 
-##### Returns
+##### Data
 
-This function returns `chainAccount(chainID)`.
+The data property follows the `terminatedStateAccountSchema` schema given [above](#json-schema-5).
 
-#### getChannel
+#### TerminatedOutboxCreated
 
-The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_OUTBOX_CREATED`. This event is emitted when a new entry is created in the terminated outbox substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated outbox account.
 
-##### Returns
+##### Data
 
-This function returns `channel(chainID)`.
+The data property follows the `terminatedOutboxAccountSchema` schema given [above](#json-schema-6).
 
+### Protocol Logic for Other Modules
 
-#### getOwnChainAccount
+#### send
 
-The `getOwnChainAccount` function returns the object stored in the own chain data substore.
+The `send` function is used to create and add a message to the outbox of a partner chain.
 
-##### Parameters
+```python
+def send(sendingAddress: Address, 
+    module: Module, 
+    crossChainCommand: CrossChainCommand, 
+    receivingChainID: ChainID, 
+    fee: uint64, 
+    status: uint32, 
+    params: object
+) -> None:
+# Create cross-chain message
+    paramsSchema = JSON schema associated with (module, crossChainCommand)
+    ccm = {
+        "nonce": ownChainAccount.nonce,
+        "module": module,
+        "crossChainCommand": crossChainCommand,
+        "sendingChainID": ownChainAccount.chainID,
+        "receivingChainID": receivingChainID,
+        "fee": fee,
+        "status": status,
+        "params": encode(paramsSchema, params)
+    }
 
-This function takes no input.
+    # Validate ccm size
+    try:
+        validateFormat(ccm)
+    except Exception as e:
+        emitPersistentEvent(
+            module = MODULE_NAME_INTEROPERABILITY,
+            name = EVENT_NAME_CCM_SENT,
+            data = {"ccmID": ccmID, "result": ERROR_CODE_INVALID_FORMAT},
+            topics = [ownChainAccount.chainID, receivingChainID, EMPTY_BYTES]
+        )
+        raise e
+
+    # From now on, we can assume that the ccm is valid.
+    ccmID = sha256(encode(crossChainMessageSchema, ccm))
 
-##### Returns
+    # Processing on the mainchain
```

Comment:
I would not call this processing, but maybe "sending" or something similar. As the processing is only done in the receiving chain.

Reply:
all good